File: 2144-maximum-difference-between-increasing-elements/maximum-difference-between-increasing-elements.py
class Solution(object):
    def maximumDifference(self, nums):
        min_val = nums[0]
        ans = -1
        for num in nums[1:]:
            if num > min_val:
                ans = max(ans, num - min_val)

            min_val = min(min_val, num)

        return ans

File: 2144-maximum-difference-between-increasing-elements/maximum-difference-between-increasing-elements.cpp
class Solution {
public:
    int maximumDifference(vector<int>& nums) {

        int min_val = nums[0];
        int ans = -1;
        for (int num : nums) {
            if (num > min_val) {
                ans = max(ans, num - min_val);
            }
            min_val = min(min_val, num);
        }
        return ans;
    }
};